src/DynamoCore/Configuration/PreferenceSettings.cs

```diff
@@ -841,8 +857,34 @@ public bool IsTrustedLocation(string path)
             {
                 string location = PathHelper.ValidateDirectory(path);
 
+                var locationDir = new DirectoryInfo(location);
                 // All subdirectories are considered trusted if the parent directory is trusted.
-                return TrustedLocations.FirstOrDefault(x => location.StartsWith(x)) != null;
+                return TrustedLocations.FirstOrDefault(x =>
+                {
+                    if (location.Equals(x))
```

Comment:
should we be using to lower invariant or something ?

Reply:
Added new PathHelper.AreDirectoryPathsEqual API